### PR TITLE
MEPTS-476: Review and Fix Transferred Out Query in TX ML

### PR DESCRIPTION
**Pull request by rubailly**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxMlCohortQueries.java [hunk 1]
```diff
@@ -361,47 +361,166 @@ public CohortDefinition getPatientsLTFUMoreThan90DaysComposition() {
...
-            "onOrBefore=${endDate},location=${location}"));
+    SqlCohortDefinition sqlCohortDefinition = new SqlCohortDefinition();
 
-    cd.addSearch(
-        "permanentStateTransferredOut",
-        EptsReportUtils.map(
-            txCurrCohortQueries
-                .getTransferredOutPatientsInFichaResumeAndClinicaOfMasterCardByReportEndDate(),
-            "onOrBefore=${endDate},location=${location}"));
+    sqlCohortDefinition.setName(
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxMlCohortQueries.java:373]: @rubailly So if this is the same as Prev and TB, why not to use the same query?

Comment (psmatsinhe) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxMlCohortQueries.java:373]: I mean all them from the same place?

Comment (rubailly) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxMlCohortQueries.java:373]: I think I heard at some point @ynurmahomed mentioning this cohort query classes should be independent as much as possible. I then found the method was duplicated in two classes and thought it was intentional.
I didn't give it much thought but to me having a parent class where they can all inherit these queries would be a better approach. Maybe we can have a discussion around this again with Yassin and come up with a conclusion.

Comment (psmatsinhe) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxMlCohortQueries.java:373]: @ningosi what do you think about this?

Comment (psmatsinhe) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxMlCohortQueries.java:373]: Let's have this merged and then we will agree on this!

Comment (ynurmahomed) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxMlCohortQueries.java:373]: @rubailly @psmatsinhe I think the only re-usable queries are in the [common queries doc ](https://docs.google.com/document/d/1EtpeIn-6seD5skZJteCdANhxkKXQye9RckGV2eoYj6c/edit?pli=1). 

If it does not point to that doc maybe better to duplicate it.

In the past I've tried re-using first then if the specs diverge duplicate it, but ended up having to duplicate the query most of the time.

**Pull request merged**